controllers/cluster_controller_phases_test.go

```diff
@@ -370,3 +370,184 @@ func TestClusterReconciler_reconcilePhase(t *testing.T) {
 		})
 	}
 }
+
+func TestClusterReconcilePhases_reconcileFailureDomains(t *testing.T) {
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-cluster",
+			Namespace: "test-namespace",
+		},
+		Status: clusterv1.ClusterStatus{
+			InfrastructureReady: true,
+		},
+		Spec: clusterv1.ClusterSpec{
+			ControlPlaneEndpoint: clusterv1.APIEndpoint{
+				Host: "1.2.3.4",
+				Port: 8443,
+			},
+			InfrastructureRef: &corev1.ObjectReference{
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				Kind:       "GenericInfrastructureMachine",
+				Name:       "test",
+			},
+		},
+	}
+
+	newFailureDomain := clusterv1.FailureDomains{
+		"newdomain": clusterv1.FailureDomainSpec{
+			ControlPlane: false,
+			Attributes: map[string]string{
+				"attribute1": "value1",
+			},
+		},
+	}
+
+	newFailureDomainUpdated := clusterv1.FailureDomains{
+		"newdomain": clusterv1.FailureDomainSpec{
+			ControlPlane: false,
+			Attributes: map[string]string{
+				"attribute2": "value2",
+			},
+		},
+	}
+
+	clusterWithnewFailureDomainDifferentAttributes := cluster.DeepCopy()
+	clusterWithnewFailureDomainDifferentAttributes.Status.FailureDomains = newFailureDomainUpdated
+
+	oldFailureDomain := clusterv1.FailureDomains{
+		"domainold": clusterv1.FailureDomainSpec{
```

Comment:
nit: consistently with the above ("newdomain") this should be "olddomain"

Reply:
Ack